### ISSUE TITLE
Rate Limits

### ISSUE THREAD

**Issue by ZincNitrogen** [4chan/4chan-API#101]
Hi, I believe i have just gotten rate limited (axios all of a sudden gives a 429 "Too Many Requests" Error when everything was working before and there have been no changes to the code). 

I read in another issue that the rate limit is 1 request per second, but I haven't been able to find that in the docs.

Does anyone know how long it takes to get unblocked?

**Comment by AaronS95** [comment]
I'm having the same issue - no changes to my code either

It works when viewing the JSON in the browser, so it seems to be still working

**Comment by ZincNitrogen** [comment]
@AaronS95 Yeah I can also confirm the endpoint works when you go through the browser. My current plan is to essentially not make any API requests til Monday and see if I get automatically unblocked.

**Comment by Hamuko** [comment]
Are you sending a `User-Agent` header with your requests?

**Comment by FrostyFridge** [comment]
Don't think you're blocked, it seems like all i.c4dn requests require a `User-Agent` header now or else you'll get a 429

**Comment by AaronS95** [comment]
Changed my code a bit and it seems like I was missing a User-Agent header, works fine now!

Before making any changes I logged the output of my request headers which did include _User-Agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10.15; rv:128.0) Gecko/20100101 Firefox/128.0_, so I'm not really sure what went wrong - either way it's fixed for me

**Comment by ZincNitrogen** [comment]
@Hamuko and @FrostyFridge , where did you find out that we need a `User-Agent ` header?

**Comment by Hamuko** [comment]
Noticed that cURL worked whereas my raw HTTP requests didn't, and I knew that cURL sends its `User-Agent`.

**Comment by ZincNitrogen** [comment]
@Hamuko ah I see. Weird that 4chan made that update all of a sudden.
